implementation/src/main/java/io/smallrye/metrics/setup/config/MetricPercentileConfiguration.java

```diff
@@ -0,0 +1,90 @@
+package io.smallrye.metrics.setup.config;
+
+import java.util.ArrayDeque;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+public class MetricPercentileConfiguration extends PropertyArrayConfiguration<Double> {
+    private boolean isDisabled = false;
+
+    private static final String CLASS_NAME = MetricPercentileConfiguration.class.getName();
+    private static final Logger LOGGER = Logger.getLogger(CLASS_NAME);
+
+    public MetricPercentileConfiguration(String metricName, Double[] percentileValues) {
+        super(metricName, percentileValues);
+    }
+
+    public MetricPercentileConfiguration(String metricName, boolean isDisabled) {
+        super(metricName, null);
+        this.isDisabled = isDisabled;
+    }
+
+    public boolean isDisabled() {
+        return isDisabled;
+    }
+
+    /**
+     * 
+     * Parse the `mp.metrics.distribution.percentile` property.
+     * syntax of <metric_name>=<value-1>,<value-2>,...,<value-n>
+     * No values supplied to a metric name disables percentile output.
+     * Can use wild card `*` at the end of metric name (e.g. demo.app.*)
+     * 
+     * @param input MP Config value
+     * @return Collection<MetricPercentileConfiguration> Collection of MetricPercentileConfiguration objects
+     */
+    public static Collection<MetricPercentileConfiguration> parseMetricPercentiles(String input) {
+
+        ArrayDeque<MetricPercentileConfiguration> metricPercentileConfiCollection = new ArrayDeque<MetricPercentileConfiguration>();
+
+        if (input == null || input.length() == 0) {
+            return null;
+        }
+
+        // not expecting backslashes?
+        String[] metricValuePairs = input.split(";");
+
+        // Individual metric name grouping and values
+        for (String kvString : metricValuePairs) {
+
+            String[] keyValueSplit = kvString.split("=");
+
+            String metricName = keyValueSplit[0];
+
+            MetricPercentileConfiguration metricPercentileConfiguration;
+
+            // empty value - disabled
+            if (keyValueSplit.length == 1) {
+                metricPercentileConfiguration = new MetricPercentileConfiguration(metricName, true);
+            } else {
+                // Parse values of percentile - ensure value is 0 <= x <= 1.0
+                Double[] percentileValues = Arrays.asList(keyValueSplit[1].split(",")).stream().map(s -> {
+
+                    if (s.matches("[0][.][0-9]+")) {
```

Comment:
Does the dot need escape?

Reply:
the `[  ]` character set allows for matching explicitly the characters within it.